### ISSUE TITLE
How can vilfredo integrate with slack.com

### ISSUE THREAD

**Issue by vvorski** [fairdemocracy/vilfredo-core#52]
slack.com is a rapidly growing collaboration-chat system. It supports a huge number of integrations https://slack.com/integrations  but nothing for democracy and decision making, I think this is worth investigating how vilfredo would integrate with a collaboration chat service like this. (there are others such as flowdock, but slack.com is hot now). 

It could be developed separately from the client and now would be a good time to get into the game as slack is growing rapidly and people are still searching for new apps to add to it.

**Comment by vvorski** [comment]
@pietrosperoni what do you think?

**Comment by pietrosperoni** [comment]
Looks like a really interesting tool. Vilfredo is designed with fairness in mind. Even putting it before speed and practicality. So people who have a lot of friends are not allowed to mail them to ask them to support their proposal, even though this might make a good proposal go viral. Because it might also kill other good proposals that did not have the luck to be written by someone very popular. Moving away from this principle completely kills la raison d'ètre of Vilfredo. So this is important. 

For this reason it took me more than 5 years to find a way to integrate simple commenting inside Vilfredo. The risk over there was that people would "take decisions" outside of the platform. And stripping away from Vilfredo the push power that it has to get the most democratic decision from being implemented. It took a lot of time, but now I am quite happy with the result. 

Vilfredo needs an instant communication system. Maybe even a voice or even video system. But it has to  appear as the right tool for what Vilfredo wants and needs to achieve. Not just as a tool in itself that we now want to integrate inside. It needs to be something controlled by the AI that as the moderator, based upon how people voted puts in touch certain people with certain other people. In this way we put the foundations for a Vilfredo which can really grow while remaining fair.

If instead we simply integrate some sort of common instant messaging system to a question, it would take away the fairness from the process generating back into the general discussion, like a room full of people shouting, we are so much trying to move away from. So I am very dubious. Of course if slack could permit to have a system outside (the AI) decide who can speak and who cannot. SO it gives to this system a strong active moderator role in the discussion then it might be useful. But I fear it would be perceived as dictatorial to people that are used to use slack and use it with no limitations. 

So I suggest to put this under the Maybe Someday section, maybe wait until a time when we need instantaneous communication. And by that time hopefully we will have tried it all, and have the idea more clear :-).

**Comment by vvorski** [comment]
Sounds like you want to engineer out the human element out of the decision making process? 

People will talk. No matter how much you might want to control free speech it will find a way.

In any case I thought we were talking about small working groups.

With 5-10 people I don't think any of the problems you fear would arise.

V/.

**Comment by vvorski** [comment]
Slack is just a communication platform, like email, nothing more. Vilfredo already uses email. This would be a better notification channel.

V/.

**Comment by pietrosperoni** [comment]
Not really, Vilfredo uses email from Vilfredo to the people, 1 to many. Not between the people. The discussion is inside Vilfredo. 
From time to time Vilfredo (the old one at least, and we still have to reprogram) puts some specific people in touch with some specific others, to speak about a specific topic. It is not a general discussion place. And all this is done to make sure that each person has an equal saying in the result. So yes, it is taking away some human elements from the process. The natural tendency of human people to organise in a hierarchical structure, for example. Giving more power to the people that speak more. This has an equivalent in the Sociocratic process, which you know and respect. When a decision is being taken, the moderator, and the moderator alone decide who speaks, for how long. And then what you do next, and so on. Introducing a way for people to chat indiscriminately inside a sociocratic discussion would harm and hinder the fairness of the discussion (even though at times it might let the people reach a consensus more rapidly). 

Also, Vilfredo is trying to break the 12 people limit, for example which is that limit that makes real collaboration in more than a dozen people quite impossible.

**Comment by giovanisp** [comment]
I imagine something like the old IRC system, where some people in the room could be given @voice , the difference is that the channel op is Vilfredo AI engine.

Inviato da iPad

**Comment by vvorski** [comment]
I think slack is tangential to this . It's a channel for pushing announcements the Vilfredo bot can send them individually to each person.

Email is a horrible method for communicating.

The future is integrated tools and slack is one promising integration hub..

V/.

**Comment by pietrosperoni** [comment]
Maybe Someday. But yes, done like that is like we would integrate it in FB. Maybe Someday too :-)
